sklearn/linear_model/tests/test_ridge.py

```diff
@@ -1234,20 +1234,61 @@ def test_ridgecv_int_alphas():
     ridge.fit(X, y)
 
 
-def test_ridgecv_negative_alphas():
+def test_ridgecv_scalar_alphas():
     X = np.array([[-1.0, -1.0], [-1.0, 0], [-0.8, -1.0], [1.0, 1.0], [1.0, 0.0]])
     y = [1, 1, 1, -1, -1]
+    # The method for fitting _BaseRidgeCV depends whether cv=None
+    cv = KFold(3)
 
     # Negative integers
-    ridge = RidgeCV(alphas=(-1, -10, -100))
-    with pytest.raises(ValueError, match="alphas must be strictly positive"):
+    ridge = RidgeCV(alphas=(1, -1, -100))
+    with pytest.raises(ValueError, match=r"alphas\[1\] == -1, must be > 0.0"):
         ridge.fit(X, y)
 
-    # Negative floats
-    ridge = RidgeCV(alphas=(-0.1, -1.0, -10.0))
-    with pytest.raises(ValueError, match="alphas must be strictly positive"):
+    # Negative floats and cv is not None
+    ridge = RidgeCV(alphas=(-0.1, -1.0, -10.0), cv=cv)
+    with pytest.raises(ValueError, match=r"alphas\[0\] == -0.1, must be > 0.0"):
         ridge.fit(X, y)
 
+    # Strings
+    ridge = RidgeCV(alphas=(1, 1.0, "1"))
+    with pytest.raises(
+        TypeError,
+        match=(
```

Comment:
you can create before `with pytest.raises` a variable `err_msg` to store the message (could be the partial message) to be matched. It will be more reasable.